packages/mui-styles/src/styled/styled.js

```diff
@@ -9,7 +9,7 @@ function omit(input, fields) {
   const output = {};
 
   Object.keys(input).forEach((prop) => {
-    if (fields.indexOf(prop) === -1) {
+    if (!fields.includes(prop)) {
```

Comment:
Just for demonstration purposes, but this is one of those hidden O(nm) complexities that can be reduced down to O(n)+O(m):

```tsx
function omit(input, fields) {
  const output = {};
  const fieldSet = new Set(fields)
  Object.keys(input).forEach((prop) => {
    if (!fieldSet.has(prop)) {
      output[prop] = input[prop];
    }
  });
  return output;
}
```

Not to be changed unless proven to be a real performance improvement though.